### PR TITLE
Customizable projects

### PR DESCRIPTION
**Pull request by jyotheeshwar**
Still a WIP

### CODE REVIEW

**Comment by jyotheeshwar** [comment]
Tested locally with a `chalice local` command - ready for review

**Comment by jyotheeshwar** [comment]
NOTE: It is required that all new projects are placed under the `project` directory. And each of them should have a file called `indexer.py` which has a class called `Indexer` that is a child of `BaseIndexer` from `utils/indexer.py`

File: app.py [hunk 1]
```diff
@@ -18,8 +18,10 @@
 # SampleOrientedIndexer as SampleIndexer, \
 # ProjectOrientedIndexer as ProjectIndexer
 from chalicelib.utils import DataExtractor
+from utils.indexer import BaseIndexer
```
Comment (caaespin) [inline app.py:21]: @jyotheeshwar Can you check if this works when deploying to AWS lambda? I know that when packaging the application and pushing to AWS, all the modules must be packed inside a folder called`chalicelib`. Does your automated deployment strategy deal with that? See:
http://chalice.readthedocs.io/en/latest/topics/packaging.html

**Review by caaespin — APPROVED** [review]
@jyotheeshwar please take a look at the comment I made. I'll got ahead and merge your branch right now, so I can incorporate the changes you made into the code I'm writing, but please do check that, and if you need to, branch out again from `feature/refactor` should you need to make other necessary changes. :)

**Pull request merged**